### ISSUE TITLE
Decorator not set in Rails 7.1.2

### ISSUE THREAD

**Issue by ryanb** [wallaby-rails/wallaby-rails#212]
I have an `accounts/_resource_actions.html.erb` partial where I reference `decorated` and it's returning an instance of `ApplicationDecorator` instead of the `AccountDecorator` like it did in Rails 7.1.1. Perhaps something changed with how loading works?

```
Failure/Error: <% if decorated.intercom_url.present? %>
          
          ActionView::Template::Error:
            undefined method `intercom_url' for #<SupportAdmin::ApplicationDecorator:0x0000000112e99950...
```

Any ideas @tian-im?

**Comment by tian-im** [comment]
@ryanb Not sure if this is related to this fix for `config.eager_load_paths` at https://github.com/rails/rails/blob/v7.1.2/railties/CHANGELOG.md#rails-712-november-10-2023.

Would you happen to have a minimum setup that I can reproduce? Otherwise, I will see how I can reproduce it in this weekend.

BTW, I've released a beta version [wallaby-7.0.0.beta1](https://rubygems.org/gems/wallaby/versions/7.0.0.beta1) that should fix most of the issues you've brought up. So sorry for keeping you waiting so long, I was hoping to get it to a state that I'm satisfied with, but I realized that I won't be able to do that and I should have broken it down into smaller releases. If you want to give it a go, let me know how it'd go 🙇

**Comment by ryanb** [comment]
@tian-im I plan to investigate this further on Friday and will upgrade Wallaby then. Thanks!

**Comment by ryanb** [comment]
@tian-im btw, is there a changelog of the 7.0 changes? It's too hard to follow the commits since there are so many repos.

**Comment by tian-im** [comment]
Ahh, forgot to update the changelog. Will do that tomorrow!

**Comment by ryanb** [comment]
@tian-im I'm unable to try Wallaby 7.0.0.beta1 with Rails 7.1.2 since it isn't compatible. See:

```
Because wallaby >= 7.0.0.beta1 depends on wallaby-active_record = 0.3.0.beta1
  and wallaby-active_record >= 0.3.0.beta1 depends on wallaby-core = 0.3.0.beta1,
  wallaby >= 7.0.0.beta1 requires wallaby-core = 0.3.0.beta1.
And because wallaby-core >= 0.3.0.beta1 depends on railties >= 4.2.0, < 7.1.0
  and rails >= 7.1.2 depends on railties = 7.1.2,
  wallaby >= 7.0.0.beta1 is incompatible with rails >= 7.1.2.
So, because Gemfile depends on rails = 7.1.2
  and Gemfile depends on wallaby = 7.0.0.beta1,
  version solving has failed.
```

**Comment by ryanb** [comment]
@tian-im here's an example Rails 7.1.2 app that doesn't work with the custom decorator. https://github.com/ryanb/wallaby-rails-7-1-2

Run `bundle`, `rails db:migrate` and `rails s` and navigate to http://localhost:3000/admin/blogs and it will complain once you create a blog record.

**Comment by tian-im** [comment]
> @tian-im here's an example Rails 7.1.2 app that doesn't work with the custom decorator. https://github.com/ryanb/wallaby-rails-7-1-2

@ryanb The fix is done at https://github.com/wallaby-rails/wallaby-core/pull/77 and gem `wallaby-core` 0.2.9 is released. Let me know how you'd go with the fix.

**Comment by tian-im** [comment]
> @tian-im I'm unable to try Wallaby 7.0.0.beta1 with Rails 7.1.2 since it isn't compatible. See:

Ahhh, forgot that Rails is already 7.1, will test wallaby 7.0.0 beta with Rails 7.1. Will let you know when it's ready again.

**Comment by ryanb** [comment]
@tian-im btw, I've had no issues with Rails 7.1.1, just 7.1.2.

**Comment by ryanb** [comment]
@tian-im any update on this? Are you able to duplicate the issue?

**Comment by tian-im** [comment]
@ryanb do you mean the fix or the beta?

If it's the fix, I've done it in `wallaby-core` 0.2.9. If you bundle update `wallaby-core`, it should fix the problem.

If it's the beta, I'm still testing it, but very close to releasing beta2. Will let you know once it's ready.

**Comment by ryanb** [comment]
@tian-im thanks, I missed that message about it being fixed in `wallaby-core`. I'll try it out.

**Comment by ryanb** [comment]
This is fixed btw, forgot to mention.

**Comment by ryanb** [comment]
@tian-im this issue has come back in 7 beta2. Any ideas?

```
NoMethodError - undefined method `intercom_url' for #<SupportAdmin::ApplicationDecorator:...>:
15:26:48 web.1     |   app/views/support_admin/accounts/_resource_actions.html.erb:1
15:26:48 web.1     |   app/controllers/concerns/slow_request_concern.rb:13:in `notify_slow_requests'
15:26:48 web.1     |   app/controllers/application_controller.rb:82:in `set_time_zone!'
```

**Comment by ryanb** [comment]
@tian-im any thoughts on this one? I think it's the only blocker on adopting 7.0 beta 2.

Thanks.

**Comment by tian-im** [comment]
@ryanb will continue with this today. Will update when it's fixed 😁

**Comment by tian-im** [comment]
@ryanb I tried to replicate the issue in this pull request to https://github.com/ryanb/wallaby-rails-7-1-2/pull/1.

I couldn't replicate the issue by either saving a new blog or changing the blog decorator. Do you have some steps that I can use to replicate this issue with `wallaby-7.0.0.beta2`?

**Comment by ryanb** [comment]
@tian-im Thanks for looking into this. Found the issue. I had this line in the `AccountDecorator`

```
alias_method :extend_trial_metadata_of, :form_metadata_of
```

Seems `form_metadata_of` method was removed so this was raising an exception. The exception was being ignored so I'll make a separate issue for that.
